### PR TITLE
chore: refactor cloud sync

### PR DESCRIPTION
**Pull request by tassoevan**
## Issue(s)

[NBJ-333](https://rocketchat.atlassian.net/browse/NBJ-333)

## Further comments

[NBJ-333]: https://rocketchat.atlassian.net/browse/NBJ-333?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ggazzo — APPROVED** [review]

File: apps/meteor/server/models/raw/CloudAnnouncements.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import { BaseRaw } from './BaseRaw';
+
+export class CloudAnnouncementsRaw extends BaseRaw<Cloud.Announcement> implements ICloudAnnouncementsModel {
+	constructor(db: Db, trash?: Collection<RocketChatRecordDeleted<Cloud.Announcement>>) {
+		super(db, 'cloud_announcements', trash);
+	}
+
+	modelIndexes(): IndexDescription[] {
+		return [{ key: { status: 1, expireAt: 1 } }];
```
Comment (sampaiodiego) [inline apps/meteor/server/models/raw/CloudAnnouncements.ts:13]: are these indexes supposed to be used any time in the future?

File: apps/meteor/server/models/raw/CloudAnnouncements.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import type { Cloud, RocketChatRecordDeleted } from '@rocket.chat/core-typings';
+import type { ICloudAnnouncementsModel } from '@rocket.chat/model-typings';
+import type { Collection, Db, IndexDescription } from 'mongodb';
+
+import { BaseRaw } from './BaseRaw';
+
+export class CloudAnnouncementsRaw extends BaseRaw<Cloud.Announcement> implements ICloudAnnouncementsModel {
+	constructor(db: Db, trash?: Collection<RocketChatRecordDeleted<Cloud.Announcement>>) {
+		super(db, 'cloud_announcements', trash);
```
Comment (sampaiodiego) [inline apps/meteor/server/models/raw/CloudAnnouncements.ts:9]: I think we won't need `thrash` for this collection

File: packages/core-typings/package.json [hunk 3]
```diff
@@ -26,6 +20,13 @@
 		"@rocket.chat/message-parser": "next",
 		"@rocket.chat/ui-kit": "^0.32.1"
 	},
+	"devDependencies": {
+		"@rocket.chat/eslint-config": "workspace:^",
+		"eslint": "~8.45.0",
+		"mongodb": "^4.12.1",
```
Comment (sampaiodiego) [inline packages/core-typings/package.json:26]: 4.17?

File: packages/core-typings/src/cloud/Announcement.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable @typescript-eslint/naming-convention */
```
Comment (sampaiodiego) [inline packages/core-typings/src/cloud/Announcement.ts:1]: should we remove this rule?

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: packages/core-typings/src/cloud/Announcement.ts (hunk 4 above)
Comment (ggazzo) [reply packages/core-typings/src/cloud/Announcement.ts:1]: please

**Review by sampaiodiego — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]

**Comment by zahidhussain998** [comment]
hey, i would like to work on this issue

**Pull request merged**